src/esp/assets/ResourceManager.cpp

```diff
@@ -1108,6 +1109,263 @@ bool ResourceManager::loadGeneralMeshData(
   return true;
 }  // loadGeneralMeshData
 
+std::vector<Mn::Vector3> ResourceManager::buildSmoothTrajOfPoints(
```

Comment:
This is a lot of logic specific to tubes/splines/geometry. Can you move it to an appropriate file and class?

Reply:
Yeah, the spline construction can probably live in `esp/geo`

Reply:
And even have nice Python bindings!